dao/src/test/kotlin/tables/PackageDeduplicationMigrationTest.kt

```diff
@@ -0,0 +1,385 @@
+
+import kotlin.time.Duration.Companion.minutes
+
+import kotlinx.datetime.Clock
+
+import org.eclipse.apoapsis.ortserver.dao.dbQuery
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.AnalyzerConfigurationDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.AnalyzerRunDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.AuthorDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.MappedDeclaredLicenseDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.PackageDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.PackagesAnalyzerRunsTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.PackagesAuthorsTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.PackagesDeclaredLicensesTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.ProcessedDeclaredLicenseDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.ProcessedDeclaredLicensesMappedDeclaredLicensesTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.ProcessedDeclaredLicensesUnmappedDeclaredLicensesTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.ProjectDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.ProjectScopeDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.ProjectsAnalyzerRunsTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.ProjectsAuthorsTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.ProjectsDeclaredLicensesTable
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.analyzer.UnmappedDeclaredLicenseDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.shared.DeclaredLicenseDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.shared.EnvironmentDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.shared.IdentifierDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.shared.RemoteArtifactDao
+import org.eclipse.apoapsis.ortserver.dao.tables.runs.shared.VcsInfoDao
+import org.eclipse.apoapsis.ortserver.dao.test.DatabaseMigrationTestExtension
+import org.eclipse.apoapsis.ortserver.model.OrtRun
+import org.eclipse.apoapsis.ortserver.model.RepositoryType
+import org.eclipse.apoapsis.ortserver.model.runs.DependencyGraphsWrapper
+import org.eclipse.apoapsis.ortserver.model.runs.Environment
+import org.eclipse.apoapsis.ortserver.model.runs.Identifier
+import org.eclipse.apoapsis.ortserver.model.runs.Package
+import org.eclipse.apoapsis.ortserver.model.runs.ProcessedDeclaredLicense
+import org.eclipse.apoapsis.ortserver.model.runs.Project
+import org.eclipse.apoapsis.ortserver.model.runs.RemoteArtifact
+import org.eclipse.apoapsis.ortserver.model.runs.VcsInfo
+
+import org.jetbrains.exposed.sql.insert
+
+class PackageDeduplicationMigrationTest : StringSpec() {
+    val extension = extension(DatabaseMigrationTestExtension("71", "72"))
+
+    init {
+        "package migration should deduplicate packages" {
```

Comment:
I find the test hard to understand because all situations are tested in a single test case. This makes it difficult to figure out which scenarios are actually tested (e.g., compared to https://github.com/boschglobal/ort-server/blob/40ac248888f421b66bc4f84a5aeec66a120f8e94/services/maintenance/src/test/kotlin/jobs/DeduplicatePackagesJobTest.kt#L147).

Also, migration tests should be in the `migration` package and be named like the migration to easily match them (see the other tests in the package). And they should not use any of the existing Exposed tables or DAOs. This is because this code will change with future migration, but has to stay constant for the migration test. Instead, they should be copied like in the other migration tests.

However, I would be fine with addressing any of those test changes in a follow-up PR.

Reply:
Renamed the class and moved it into the `migration` package.